api/compaction_manager.cc

```diff
@@ -74,13 +74,14 @@ void set_compaction_manager(http_context& ctx, routes& r) {
 
     cm::get_pending_tasks_by_table.set(r, [&ctx] (std::unique_ptr<request> req) {
         return ctx.db.map_reduce0([&ctx](database& db) {
-            std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash> tasks;
+            return do_with(std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>(), [&ctx, &db](std::unordered_map<std::pair<sstring, sstring>, uint64_t, utils::tuple_hash>& tasks) {
             return do_for_each(db.get_column_families(), [&tasks](const std::pair<utils::UUID, seastar::lw_shared_ptr<table>>& i) {
                 table& cf = *i.second.get();
                 tasks[std::make_pair(cf.schema()->ks_name(), cf.schema()->cf_name())] = cf.get_compaction_strategy().estimated_pending_compactions(cf);
                 return make_ready_future<>();
             }).then([&tasks] {
-                return tasks;
+                return std::move(tasks);
+            });
```

Comment:
Is this really needed?

Reply:
@nyh thinks it is and he's probably right